src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx

```diff
@@ -17,16 +19,71 @@ interface Props {
 }
 
 const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree }: Props) => {
+  const { t } = useTranslation();
+  const { classes } = useStyles();
   const [failureModeDialogOpen, setFailureModeDialogOpen] = useState(false);
 
   const [failureModeOverviewDialogOpen, setFailureModeOverviewDialogOpen] = useState(false);
   const [failureModeOverview, setFailureModeOverview] = useState<FailureMode | null>(null);
 
+  const [criticality, setCriticality] = useState<number | undefined>(undefined);
+  const [predictedFailureRate, setPredictedFailureRate] = useState<number | undefined>(undefined);
+  const [ataSystem, setAtaSystem] = useState<string | undefined>(undefined);
+  const [partNumber, setPartNumber] = useState<string | undefined>(undefined);
+  const [stock, setStock] = useState<string | undefined>(undefined);
+  const [quantity, setQuantity] = useState<number | undefined>(undefined);
+
   const handleFailureModeClicked = (failureMode: FailureMode) => {
     setFailureModeOverview(failureMode);
     setFailureModeOverviewDialogOpen(true);
   };
 
+  useEffect(() => {
+    if (shapeToolData?.supertypes?.criticality) {
+      setCriticality(shapeToolData.supertypes.criticality);
+    } else {
+      setCriticality(undefined);
+    }
+
+    const types = asArray(shapeToolData?.supertypes?.supertypes)
```

Comment:
@kostobog can't we just shape it from server response so it would be always array?

Reply:
Will be solved in https://github.com/kbss-cvut/fta-fmea-ui/issues/263